components/context.tsx

```diff
@@ -0,0 +1,42 @@
+import React, { useState, useContext, FC, ReactNode } from 'react';
+
+interface Sidebar {
+  isSidebarOpen: boolean;
+  openSidebar: () => void;
+  closeSidebar: () => void;
+}
+
+interface AppProviderProps {
+  children: ReactNode | ReactNode[];
+}
+
+const AppContext = React.createContext<Sidebar | null>(null);
+
+const AppProvider: FC<AppProviderProps> = ({ children }) => {
+  const [isSidebarOpen, setIsSidebarOpen] = useState(true);
```

Comment:
While this may have been for testing, I imagine we'll want the sidebar to be closed by default:
```suggestion
  const [isSidebarOpen, setIsSidebarOpen] = useState(false);
```